### PR TITLE
Feature/replace my project page

### PR DESCRIPTION
**Pull request by ulferts**
Applies the grid module to the overview page and by that replaces the my_project_page module completely.

https://community.openproject.com/projects/openproject/work_packages/29369
https://community.openproject.com/projects/openproject/work_packages/29356
https://community.openproject.com/projects/openproject/work_packages/29370

It also disables the dashboard for all envs except 'test'. The dashboard is to be enabled again in 10.1 so I simply left the code in there but removed the necessary permissions on initialization.

#### Todo

* [x] Define default setup
* [x] Handle user opening overview page without it existing with the user lacking the `manage_overview` permission
* [x] Prevent multiple overview pages existing for the same project (second one created via the API)
* [x] Migrate existing data
* [x] Apply seeder from my_project_page to overview
* [x] Apply patch to TextileConverter from my_project_page to overview
* [x] Refactor duplicated code (e.g. rails controller, grid_configuration)
* [x] Remove my_project_page code and code in core which is referenced only by the module

### CODE REVIEW

**Comment by ulferts** [comment]
I attempted to bisect the failing spec 
```
rspec ./spec/models/work_package/work_package_action_mailer_spec.rb:55
```
but to no avail. Will have to tackle it another time.

File: modules/overviews/db/migrate/20190826083604_my_project_page_to_grid.rb [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+                  'work_packages_watched'
+               :build_wp_table_widget
+             when 'project_description',
+                  'work_package_tracking',
+                  'members',
+                  'news_latest',
+                  'calendar',
+                  'spent_time',
+                  'subprojects'
+               :build_default_widget
```
Comment (oliverguenther) [inline modules/overviews/db/migrate/20190826083604_my_project_page_to_grid.rb:89]: This fails for the `QA edge` dump due to the `widget_config` being 'documents'. Adding this in the last when case works, but maybe this is a case for adding a default case instead?

**Review by oliverguenther — APPROVED** [review]
LGTM on QA , community and some other dumps I have tested with the one comment exception on QA (documents widget). Great work and good to merge once the comment is addressed :+1:

**Comment by ulferts** [comment]
Done as proposed @oliverguenther. Thanks for catching that.

**Pull request merged**